examples/math/custom_primitives.rs

```diff
@@ -0,0 +1,496 @@
+
+// Rotate the 3D shapes.
+fn rotate_3d_shapes(mut shapes: Query<&mut Transform, With<Shape3d>>, time: Res<Time>) {
+    let delta_seconds = time.delta_seconds();
+
+    for mut transform in shapes.iter_mut() {
+        transform.rotate_y(delta_seconds);
+    }
+}
+
+// Draw the AABBs or bounding spheres for the 3D shapes.
+fn bounding_shapes_3d(
+    shapes: Query<&Transform, With<Shape3d>>,
+    mut gizmos: Gizmos,
+    bounding_shape: Res<State<BoundingShape>>,
+) {
+    for transform in shapes.iter() {
+        match bounding_shape.get() {
+            BoundingShape::None => (),
+            BoundingShape::BoundingBox => {
+                // Get the AABB of the extrusion with the rotation and translation of the mesh.
+                let aabb = EXTRUSION.aabb_3d(transform.translation, transform.rotation);
+
+                gizmos.primitive_3d(
+                    &Cuboid::from_size(Vec3::from(aabb.half_size()) * 2.),
+                    aabb.center().into(),
+                    Quat::IDENTITY,
+                    WHITE,
+                );
+            }
+            BoundingShape::BoundingSphere => {
+                // Get the bounding sphere of the extrusion with the rotation and translation of the mesh.
+                let bounding_sphere =
+                    EXTRUSION.bounding_sphere(transform.translation, transform.rotation);
+
+                gizmos.sphere(
+                    bounding_sphere.center().into(),
+                    Quat::IDENTITY,
+                    bounding_sphere.radius(),
+                    WHITE,
+                );
+            }
+        }
+    }
+}
+
+// Switch to the next bounding shape.
+fn update_bounding_shape(
+    current: Res<State<BoundingShape>>,
+    mut next: ResMut<NextState<BoundingShape>>,
+) {
+    next.set(match current.get() {
+        BoundingShape::None => BoundingShape::BoundingBox,
+        BoundingShape::BoundingBox => BoundingShape::BoundingSphere,
+        BoundingShape::BoundingSphere => BoundingShape::None,
+    });
+}
+
+// Switch between 2D and 3D cameras.
+fn switch_cameras(
+    current: Res<State<CameraActive>>,
+    mut next: ResMut<NextState<CameraActive>>,
+    mut camera: Query<(&mut Transform, &mut Projection)>,
+) {
+    let next_state = match current.get() {
+        CameraActive::Dim2 => CameraActive::Dim3,
+        CameraActive::Dim3 => CameraActive::Dim2,
+    };
+    next.set(next_state);
+
+    let (mut transform, mut projection) = camera.single_mut();
+    match next_state {
+        CameraActive::Dim2 => {
+            *transform = TRANSFORM_2D;
+            *projection = PROJECTION_2D;
+        }
+        CameraActive::Dim3 => {
+            *transform = TRANSFORM_3D;
+            *projection = PROJECTION_3D;
+        }
+    };
+}
+
+/// A custom 2D heart primitive. The heart is made up of two circles centered at `Vec2::new(±radius, 0.)` each with the same `radius`.
+/// The tip of the heart connects the two circles at a 45° angle from `Vec3::NEG_Y`.
```

Comment:
The top of the heart meets at a pretty narrow angle, approaching 0° as the resolution of the rendered image goes up. From a quick survey of random images, it looks like the angle should be 60-90°, or between 1/2 to the same as the bottom tip angle. I prefer using the same angle, but it seems up to preference.

Reply:
Yeah, I guess those are more typical, but that would further complicate a coupe of things like the `Measured2d` and meshing implementations. For a simple showcase,  I hope this can suffice :)